modules/aws/lb/main.tf

```diff
@@ -0,0 +1,301 @@
+###########################
+# Provider Configuration
+###########################
+terraform {
+  required_version = ">= 1.0.0"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.0.0"
+    }
+  }
+}
+
+###########################
+# Locals
+###########################
+
+locals {
+  is_application = var.load_balancer_type == "application"
+  is_network     = var.load_balancer_type == "network"
+}
+
+###########################################################
+# Load Balancer
+###########################################################
+
+resource "aws_lb" "load_balancer" {
+  # Common settings
+  name                       = var.name
+  internal                   = var.internal
+  load_balancer_type         = var.load_balancer_type
+  subnets                    = var.subnets
+  enable_deletion_protection = var.enable_deletion_protection
+  customer_owned_ipv4_pool   = var.customer_owned_ipv4_pool
+  ip_address_type            = var.ip_address_type
+
+  # Application Load Balancer specific settings
+  security_groups            = local.is_application ? var.security_groups : null
+  desync_mitigation_mode     = local.is_application ? var.desync_mitigation_mode : null
+  idle_timeout               = local.is_application ? var.idle_timeout : null
+  drop_invalid_header_fields = local.is_application ? var.drop_invalid_header_fields : null
+  enable_http2               = local.is_application ? var.enable_http2 : null
+  enable_waf_fail_open       = local.is_application ? var.enable_waf_fail_open : null
+
+  # Network Load Balancer specific settings
+  enable_cross_zone_load_balancing = local.is_network ? var.enable_cross_zone_load_balancing : null
+
+  dynamic "access_logs" {
+    for_each = var.access_logs != null ? { create = var.access_logs } : {}
+    content {
+      bucket  = access_logs.value.bucket
+      prefix  = access_logs.value.prefix
+      enabled = access_logs.value.enabled
+    }
+  }
+
```

Comment:
Add `dynamic "connection_logs"` block

Reply:
mirrored connection logs, but referenced the local to verify application lb in count logic.